impl/client/src/main/javascript/web/pentaho/type/action/base.js

```diff
@@ -51,10 +51,43 @@ define([
     var finishedStates = States.did | rejectedStates;
     var cancelableStates = States.init | States.will;
 
-    return Element.extend(/** @lends  pentaho.type.action.Base# */{
-      type: {
+    return Element.extend(/** @lends pentaho.type.action.Base# */{
+      type: /** @lends pentaho.type.action.Base.Type# */{
         id: module.id,
-        isAbstract: true
+
+        isAbstract: true,
+
+        // region Attribute isSync
+
+        // TODO: hierarchy consistency? immutability?
+
+        __isSync: true,
+
+        get isSync() {
+          return this.__isSync;
+        },
+
+        set isSync(value) {
+          this.__isSync = !!value;
+        },
+        // endregion
+
+        /**
+         * Performs the default action of this action type on the given action instance.
+         *
+         * When the action is [asynchronous]{@link pentaho.type.action.Base.Type#isSync},
+         * this method _may_ return a promise. If the promise is rejected,
+         * the action is rejected with the rejection reason.
+         * However, if the promise is fulfilled, its value is always *ignored*.
+         *
+         * @param {pentaho.type.action.Base} action - The action.
+         *
+         * @return {?Promise} - A promise for the completion of default action of an asynchronous action, or `null`.
+         */
+        defaultAction: function(action) {
```

Comment:
And, it exists for implementing the actions' default behavior. Check SelectAction's implementation.